ios/browser/api/password/brave_password_api.mm

```diff
@@ -0,0 +1,351 @@
+  password_store_->AddLogin([self createCredentialForm:passwordForm]);
+}
+
+- (password_manager::PasswordForm)createCredentialForm:
+    (IOSPasswordForm*)passwordForm {
+  // Store a PasswordForm representing a PasswordCredential.
+  password_manager::PasswordForm passwordCredentialForm;
+
+  passwordCredentialForm.url = net::GURLWithNSURL(passwordForm.url);
+
+  if ([passwordForm.signOnRealm length] != 0) {
+    passwordCredentialForm.signon_realm =
+        base::SysNSStringToUTF8(passwordForm.signOnRealm);
+  } else {
+    passwordCredentialForm.signon_realm = passwordCredentialForm.url.spec();
+  }
+
+  if (passwordForm.usernameElement) {
+    passwordCredentialForm.username_element =
+        base::SysNSStringToUTF16(passwordForm.usernameElement);
+  }
+
+  if (passwordForm.usernameValue) {
+    passwordCredentialForm.username_value =
+        base::SysNSStringToUTF16(passwordForm.usernameValue);
+  }
+
+  if (passwordForm.passwordElement) {
+    passwordCredentialForm.password_element =
+        base::SysNSStringToUTF16(passwordForm.passwordElement);
+  }
+
+  if (passwordForm.passwordValue) {
+    passwordCredentialForm.password_value =
+        base::SysNSStringToUTF16(passwordForm.passwordValue);
+  }
+
+  if (passwordForm.dateCreated) {
+    passwordCredentialForm.date_created =
+        base::Time::FromNSDate(passwordForm.dateCreated);
+  } else {
+    passwordCredentialForm.date_created = base::Time::Now();
+  }
+
+  if (passwordForm.dateLastUsed) {
+    passwordCredentialForm.date_last_used =
+        base::Time::FromNSDate(passwordForm.dateLastUsed);
+  } else {
+    passwordCredentialForm.date_last_used = base::Time::Now();
+  }
+
+  if (passwordForm.datePasswordChanged) {
+    passwordCredentialForm.date_password_modified =
+        base::Time::FromNSDate(passwordForm.datePasswordChanged);
+  } else {
+    passwordCredentialForm.date_password_modified = base::Time::Now();
+  }
+
+  if (passwordForm.usernameValue && !passwordForm.passwordValue) {
+    passwordCredentialForm.scheme =
+        password_manager::PasswordForm::Scheme::kUsernameOnly;
+  } else {
+    passwordCredentialForm.scheme =
+        password_manager::PasswordForm::Scheme::kHtml;
+  }
+
+  return passwordCredentialForm;
+}
+
+- (void)removeLogin:(IOSPasswordForm*)passwordForm {
+  password_store_->RemoveLogin([self createCredentialForm:passwordForm]);
+}
+
+- (void)updateLogin:(IOSPasswordForm*)newPasswordForm
+    oldPasswordForm:(IOSPasswordForm*)oldPasswordForm {
+  password_store_->UpdateLoginWithPrimaryKey(
+      [self createCredentialForm:newPasswordForm],
+      [self createCredentialForm:oldPasswordForm]);
+}
+
+- (void)getSavedLogins:
+    (void (^)(NSArray<IOSPasswordForm*>* results))completion {
+  auto fetchCredentialsCallback =
+      ^(std::vector<std::unique_ptr<password_manager::PasswordForm>> logins) {
+        NSArray<IOSPasswordForm*>* credentials =
+            [self onLoginsResult:std::move(logins)];
```

Comment:
Do we need weak self here?

Reply:
Fixed at https://github.com/brave/brave-core/pull/12360/commits/60cb0e4eae30ffd3478b2e4601c6c8f4afae59fe